### ISSUE TITLE
Layer CRS is changed even if cancel is pressed in layer properties

### ISSUE THREAD

**Issue by uclaros** [qgis/QGIS#39556]
**Describe the bug**

When a CRS is selected in Layer Properties > Source > Geometry and Coordinate Reference Systems, it is applied to the layer even if the Layer Properties dialog is closed with esc or cancel.

**How to Reproduce**
- Add a layer
- Go to Layer Properties > Source > Geometry and Coordinate Reference Systems and select a different CRS
- Press esc or click cancel
- The layer is now set to the new CRS

**QGIS and OS versions**
69faea81d9 on Debian testing

**Comment by roya0045** [comment]
Did you double click on anything?  If so this line might be the case https://github.com/qgis/QGIS/blob/194a5176ec98e41a590ced3444e50d180aaefa72/src/gui/qgsprojectionselectiondialog.cpp#L38

Unsure if this is a good thing or not. Otherwise this seems to be the offender https://github.com/qgis/QGIS/blob/194a5176ec98e41a590ced3444e50d180aaefa72/src/gui/qgsprojectionselectionwidget.cpp#L185 as it assigns the crs.

**Comment by uclaros** [comment]
> Unsure if this is a good thing or not.


The *double click on projection to accept* is super handy, I always miss it on arcmap!

The issue is actually in 
https://github.com/qgis/QGIS/blob/35d62ab0d47fa309b2de8fdad3aab66836b32585/src/gui/vector/qgsvectorlayerproperties.cpp#L108
https://github.com/qgis/QGIS/blob/35d62ab0d47fa309b2de8fdad3aab66836b32585/src/gui/vector/qgsvectorlayerproperties.cpp#L898-L904

I think the signal should be disconnected and the contents of this method be moved to `QgsVectorLayerProperties::apply()`...
Or maybe only move
```
   mLayer->setCrs( crs ); 
   mMetadataFilled = false; 
   mMetadataWidget->crsChanged(); 
```
to `QgsVectorLayerProperties::apply()`

**Comment by roya0045** [comment]
I think this is due to the fact that the selector is volatile, at least that's my experience with it, and this is why I have to use signals also to properly collect the crs.

In this case an option may be to keep the old crs somewhere and restore it upon cancelling.

**Comment by uclaros** [comment]
Fixed by #46556
